tests/specialization_constants/specialization_constants_common.h

```diff
@@ -68,6 +68,23 @@ class no_def_cnstr {
   }
 };
 
+template <typename T>
+union remove_initialization {
```

Comment:
Since this is is subtle C++ code, you can also add some comments to the code by getting inspiration along https://github.com/keryell/teaching_C-plus-plus/blob/master/non-initializing/remove_initialization.cpp

Reply:
i don't know what additional information will be useful in this case.
i've add small additional information in this commit: eea81615434077a21d3c54459d2d15c6c5c6bdaa